server/src/main/java/com/hedera/block/server/mediator/LiveStreamMediatorImpl.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Hedera Block Node
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.mediator;
+
+import com.hedera.block.protos.BlockStreamServiceGrpcProto;
+import com.hedera.block.server.consumer.LiveStreamObserver;
+import com.hedera.block.server.persistence.BlockPersistenceHandler;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * LiveStreamMediatorImpl is the implementation of the StreamMediator interface.  It is responsible for managing
+ * the subscription and unsubscription operations of downstream consumers.  It also proxies new blocks
+ * to the subscribers as they arrive and persists the blocks to the block persistence store.
+ *
+ */
+public class LiveStreamMediatorImpl implements StreamMediator<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> {
+
+    private final Logger LOGGER = Logger.getLogger(getClass().getName());
+    private final Set<LiveStreamObserver<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse>> subscribers = Collections.synchronizedSet(new LinkedHashSet<>());
```

Comment:
Rather than a synchronized set, it might be better to use a concurrent set (e.g. `ConcurrentSkipListSet` or `ConcurrentHashMap.KeySetView`) if access from multiple threads is a concern.

Reply:
My main concern with this data structure is ensuring that each subscriber gets the blocks in the order that they arrive.  Imagine a scenario like this:
```
1) notifyAll() gets called with Block A, it starts iterating through the collection of subscribers
2) Subscribers are added/removed 
3) notifyAll() gets called with Block B before Block A has been sent to every subscriber, it starts iterating through the collection of subscribers but the order is different because of changes in step 2 or the thread just completes before the Block A thread.
4) Some of the subscribers received Block B before Block A.
```
This type of scenario makes me think I should be more rigid about blocking:  https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#synchronizedSet-java.util.Set-

Reply:
Given the change in direction we discussed this morning, we will probably not need a data structure like this after the refactor.  I'll leave this as it is right now for the PR and we can resume the discussion if need be.